targets/XPlatCppSdk/source/code/source/playfab/PlayFabAndroidHttpPlugin.cpp

```diff
@@ -0,0 +1,743 @@
+        std::shared_ptr<RequestTask> requestTask = nullptr;
+        try
+        {
+            requestTask = std::make_shared<RequestTask>();
+            requestTask->Initialize(requestContainer);
+        }
+        catch (const std::exception& ex)
+        {
+            PlayFabPluginManager::GetInstance().HandleException(ex);
+        }
+        catch (...)
+        {
+
+        }
+        if(requestTask != nullptr)
+        { // LOCK httpRequestMutex
+            std::unique_lock<std::mutex> lock(httpRequestMutex);
+            requestTask->state = RequestTask::Pending;
+            pendingRequests.push_back(std::move(requestTask));
+            if(workerThread == nullptr)
+            {
+                threadRunning = true;
+                workerThread = std::make_unique<std::thread>(&PlayFabAndroidHttpPlugin::WorkerThreadEntry, this);
+            }
+        } // UNLOCK httpRequestMutex
+    }
+
+    size_t PlayFabAndroidHttpPlugin::Update()
+    {
+        if (PlayFabSettings::threadedCallbacks)
+        {
+            throw std::runtime_error("You should not call Update() when PlayFabSettings::threadedCallbacks == true");
+        }
+
+        std::shared_ptr<RequestTask> requestTask = nullptr;
+        { // LOCK httpRequestMutex
+            std::unique_lock<std::mutex> lock(httpRequestMutex);
+            if (pendingResults.empty())
+            {
+                return pendingRequests.size();
+            }
+
+            requestTask = std::move(this->pendingResults[0]);
+            this->pendingResults.pop_front();
+        } // UNLOCK httpRequestMutex
+
+        HandleResults(*requestTask);
+
+        // activeRequestCount can be altered by HandleResults, so we have to re-lock and return an updated value
+        { // LOCK httpRequestMutex
+            std::unique_lock<std::mutex> lock(httpRequestMutex);
+            return pendingRequests.size() + pendingRequests.size();
+        }
+    }
+
+    void PlayFabAndroidHttpPlugin::WorkerThreadEntry()
+    {
+        JavaVM* jvm;
+        if(!GetHelper().GetJavaVM(&jvm))
+        {
+            std::unique_lock<std::mutex> lock(httpRequestMutex);
+            threadRunning = false;
+            workerThread->detach();
+            workerThread = nullptr;
+            return ;
+        }
+        else
+        {
+            JNIEnv* jniEnv = nullptr;
+            jvm->AttachCurrentThread(&jniEnv, nullptr);
+
+            WorkerThread(static_cast<void*>(jniEnv));
+
+            jvm->DetachCurrentThread();
+        }
+    }
+
+    void PlayFabAndroidHttpPlugin::WorkerThread(void* jniEnv)
+    {
+        { // LOCK httpRequestMutex
+            std::unique_lock<std::mutex> lock(httpRequestMutex);
+            this->requestingTask = nullptr;
+        }
+        while (threadRunning) {
+            RequestTask::State state;
+            { // LOCK httpRequestMutex
+                std::unique_lock<std::mutex> lock(httpRequestMutex);
+                if (this->requestingTask == nullptr) {
```

Comment:
>{ [](start = 53, length = 1)

Please follow the current convention of code braces on next line. there are other places in code which needs to be changed in this aspect

Reply:
Ok, It was changed by Android Studio automatically. I tried to disable the option, but I couldn't failed.

---
In reply to: [274692123](https://github.com/PlayFab/SDKGenerator/pull/470#discussion_r274692123) [](ancestors = 274692123)